rclpy/src/rclpy/serialization.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Include pybind11 before rclpy_common/handle.h includes Python.h
+#include <pybind11/pybind11.h>
+
+#include <rcl/error_handling.h>
+#include <rcl/rcl.h>
+#include <rcl/types.h>
+#include <rcutils/allocator.h>
+#include <rcutils/error_handling.h>
+
+#include <memory>
+#include <stdexcept>
+
+extern "C"
+{
+#include "rclpy_common/common.h"
+}
+#include "rclpy_common/handle.h"
+
+#include "rclpy_common/exceptions.hpp"
+
+#include "serialization.hpp"
+
+namespace rclpy
+{
+class SerializedMessage
+{
+public:
+  SerializedMessage()
+  {
+    rcl_msg_ = rmw_get_zero_initialized_serialized_message();
+  }
+
+  void
+  init(rcutils_allocator_t allocator)
```

Comment:
@sloretz why a two-stage construction? Why not merging `SerializedMessage::init` with the constructor?

Reply:
I think I had it as two-stage originally so I could reuse it in the deserialize method, but ended up not going with it to avoid a copy. I merged init and the constructor in d2d3c29efb905260ff6d8542244c57a5e8bd562e